app/Console/Commands/PostcheckGAT7330.php

```diff
@@ -0,0 +1,238 @@
+            }
+            elseif (in_array($entityType, [DatasetVersion::class])) {
+                $entitiesOfThisType = $entityType::orderBy('id')->select('id')->get();
+            }
+            else {
+                $entitiesOfThisType = $entityType::orderBy('id')->select(['id', 'team_id', 'user_id'])->get();
+            }
+            foreach ($entitiesOfThisType as $entity) {
+                $csvDataRow = ['type' => null, 'id' => null, 'Collection' => null, 'Dur' => null, 'DatasetVersion' => null, 'Publication' => null, 'Tool' => null];
+                $csvDataRow['type'] = $this->classnameFromClass($entityType);
+                $csvDataRow['id'] = $entity->id;
+                $csvDataRow['team_id'] = $entity->team_id ?? null;
+                $csvDataRow['user_id'] = $entity->user_id ?? null;
+
+                foreach ($hasRelations[$this->classnameFromClass($entityType)] as $relation => $idName) {
+                    $relationEntries = array_column(
+                        (new ('App\\Models\\' . $relation)())
+                        ::where($this->classToIdString($this->classnameFromClass($entityType)), $entity->id)
+                        ->select($idName)
+                        ->get()->toArray(), $idName);
+                    
+                    $relatedEntityType = $this->idStringToClass($idName);
+
+                    if ($relatedEntityType != 'DatasetVersion') {
+                        $relatedActiveEntities = (new ('App\\Models\\' . $relatedEntityType))::whereIn('id', $relationEntries)->where('status', 'ACTIVE')->select('id')->get()->toArray();
+                    } else
+                    {
+                        $datasetVersions = (new ('App\\Models\\' . $relatedEntityType))::whereIn('id', $relationEntries)->select('id', 'dataset_id')->get()->toArray();
+                        $active = [];
+                        foreach ($datasetVersions as $datasetVersion) {
+                            $dataset = Dataset::where('id', $datasetVersion['dataset_id'])->first();
+                            if ($dataset->status = ['ACTIVE']) {
+                                $active[] = $datasetVersion;
+                            }
+                        }
+                        $relatedActiveEntities = $active;
+                    }
+                    $relatedActiveEntityIds = array_column($relatedActiveEntities, 'id');
+                    sort($relatedActiveEntityIds);
+
+                    $data[] = [$this->classnameFromClass($entityType), $entity->id, $relation, count($relatedActiveEntities), json_encode($relatedActiveEntityIds)];
+                    $csvDataRow[$relatedEntityType] = (is_null($relatedActiveEntityIds) || empty($relatedActiveEntityIds)) ? null : json_encode($relatedActiveEntityIds);
+                }
+                $csvData[] = $csvDataRow;
+            }
+        }
+
+        // $this->table($headerRow, $data);
```

Comment:
Left here for now as a simple way to change the output generated. Can't be bothered to make a more robust solution for a run-once script. Same for in the post-check script.